### PR TITLE
feat(artifact/follow): add new command "artifact follow"

### PR DESCRIPTION
**Pull request by alacuku**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> /kind flaky-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area library

/area cli

> /area tests

> /area examples

**What this PR does / why we need it**:

This PR adds the `artifact follow` command:
```bash
❯ ./falcoctl artifact follow --help
Follow a list of artifacts from remote registry. It periodically
checks if the artfacts changed and downloads the latest version based on the configured
tags.

Artifacts are passed as arguments. By just providing the name of the artifact the
command will search for the artifact in the configured index files (see index command).
If found it will use the 'registry' and 'repository' as specifiend in the index file.

Example - Install and follow "cloudtrail" plugin using the "latest" (default) tag using the info found in the index file:
        falcoctl artifact follow cloudtrail

Example - Install and follow "cloudtrail:0.6.0" plugin using the "0.6.0" tag. Here we explicitly set the tag:
        falcoctl artifact follow cloudtrail:0.6.0

Example - Install and follow "cloudtrail" plugin and "cloutrail-rules" using the "latest" (default) tag:
        falcoctl artifact follow cloudtrail cloudtrail-rules


The command also supports the references for the artifacts composed by "registry" + "repository" + "tag":

Example - Install and follow "cloudtrail" plugins using the full artifact reference:
        falcoctl artifact follow ghcr.io/falcosecurity/plugins/ruleset/cloudtrail:0.6.0-rc1

Usage:
  falcoctl artifact follow [ref1 [ref2 ...]] [flags]

Flags:
  -e, --every duration          Time interval how often it checks for a new version of the artifact (default 24h0m0s)
  -h, --help                    help for follow
      --plugins-dir string      Directory where to install plugins (default "/usr/share/falco/plugins")
      --rulesfiles-dir string   Directory where to install rules (default "/etc/falco")

Global Flags:
      --disable-styling   Disable output styling such as spinners, progress bars and colors. Styling is automatically disabled if not attacched to a tty (default false)
  -v, --verbose           Enable verbose logs (default false)
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by leogr — APPROVED** [review]
LGTM!

/approve

Waiting for a second opinion cc @falcosecurity/falcoctl-maintainers 
/hold

**Review by FedeDP — APPROVED** [review]
I love this.
+1 from me @leogr 
/approve

**Review by LucaGuerra — APPROVED** [review]
/unhold

**Pull request merged**
